src/control/control.h

```diff
@@ -183,8 +183,29 @@ class ControlDoublePrivate : public QObject {
 
     const ConfigKey m_key;
 
+    QSharedPointer<ControlNumericBehavior> m_pBehavior;
+
+    // User-visible, i18n name for what the control is.
+    QString m_name;
+
+    // User-visible, i18n description for what the control does.
+    QString m_description;
+
+    // The control value.
+    ControlValueAtomic<double> m_value;
+    // The default control value.
+    ControlValueAtomic<double> m_defaultValue;
```

Comment:
How do we protect that for get cluttered again? Maybe define regions via comments?

Reply:
We could try `-Wpadded` https://developers.redhat.com/blog/2016/06/01/how-to-avoid-wasting-megabytes-of-memory-a-few-bytes-at-a-time#tools

Reply:
I tried -Wpadded on CI, but it does not only warn about padding inbetween the data structure, but also if the overall size is not matching the byte alignment size, e.g. any class with 6 bytes would need to filled up with 2 explicit padding bytes.

I found also a currently unused macro in https://github.com/mixxxdj/mixxx/blob/15b4444584b9cd2514946d6a645746cae3f5a315/src/util/platform.h#L15

Maybe someone on Linux could try: https://linux.die.net/man/1/pahole

Reply:
Okay that makes sense I guess when the compiler expects you to nest these packed data structures. The only other potential solution I found was to generate two versions of the type, one packed (potentially with suboptimal alignment and then `static_assert(sizeof(T) == sizeof(_T_packed)) (potentially with a bit of rounding to manually account for the overall size). I wouldn't know how to accomplish that without copy pasting or a macro (doing the copy-pasting automatically) though.

Reply:
Let's not over engineer it. Just a comment state informs there reader that the oder of the variables has been chosen to avoid memory gaps due to alignment is probably sufficient.

Reply:
It's already added: https://github.com/mixxxdj/mixxx/blob/6728c9156caaa904698da5cef3f723279a5129ed/src/control/control.h#L203-L205